lib/matplotlib/tests/test_backend_pgf.py

```diff
@@ -406,3 +406,27 @@ def test_sketch_params():
     # \pgfdecoratecurrentpath must be after the path definition and before the
     # path is used (\pgfusepath)
     assert baseline in buf
+
+
+# test to make sure that the document font size is set consistently (see #26892)
+@needs_pgf_xelatex
+@pytest.mark.skipif(
+    not _has_tex_package('unicode-math'), reason='needs unicode-math.sty'
+)
+@pytest.mark.backend('pgf')
+@image_comparison(['pgf_document_font_size.pdf'], style='default', remove_text=True)
+def test_document_font_size():
```

Comment:
1) do you really need unicode-math for this test?  I'd say it's not particularly relevant, no?
2) you could do without the plot at all and simply use figtext() to draw text on an empty figure.

Reply:
1. Not `unicode-math` specifically, no. But if I remember correctly, `unicode-math` (and other math font packages) select the font variant they use at package load time based on the document font size, and these variants have different metrics even for the same font size, which is what caused the original issue. So I’d either need `unicode-math` or some random font package with the same behavior, and `unicode-math` seems much more universal (especially since matplotlib already uses it by default).
2. I suppose I could. I just took the easiest path and used the example I already had in #26892. Let me know if you really want me to make and test a new example.

Reply:
Actually, I think I was mistaken with my comment that matplotlib uses `unicode-math` (I was thinking of `fontspec`). Nevertheless, the rest still applies.